pkg/apis/resolution/v1alpha1/resolution_request_types.go

```diff
@@ -78,9 +78,19 @@ type ResolutionRequestStatusFields struct {
 	// of the requested resource in-lined into the ResolutionRequest
 	// object.
 	Data string `json:"data"`
+	// Source is the source reference of the remote data that records the url, digest
+	// and the entrypoint.
+	Source *ConfigSource `json:"source"`
 }
 
 // GetStatus implements KRShaped.
 func (rr *ResolutionRequest) GetStatus() *duckv1.Status {
 	return &rr.Status.Status
 }
+
+// ConfigSource are the source reference of the remote resource in the standard SLSA format.
+type ConfigSource struct {
+	URI        string            `json:"uri,omitempty"`
+	Digest     map[string]string `json:"digest,omitempty"`
```

Comment:
As we are migrating from Tekton Hub to Artifact Hub, we plan to add a new parameter like ```source``` in the Hub resolver to specify the source of resource. In the migration period we will support both Tekton Hub and Artifact Hub as sources. As the community agreed to deprecate  [Tekton Hub](https://hub.tekton.dev/) in the future, we may drop the Tekton Hub support of the resolver.

In this case, do we still want to support ```ConfigSources``` from both Tekton Hub and Artifact Hub (Tekton Hub currently does not return Digest in the API)?

Reply:
Thanks @QuanZhang-William for following up on this topic here.

The digest we need here is the sha digest at the moment of fetching the remote resource. For directory-based versioning, the actual sha mapped to a version might not be deterministic i.e. a new commit is pushed to that version directory. So what we need is the latest sha at the time of resolving in that version directory.

If tekton hub will be deprecated anyways, I think we can jut focus on enabling Artifact Hub resolver to set the accurate value for the configsource. 

Thoughts? 

cc @dibyom  @wlynch

Reply:
For directory based catalog you'd key off of the commit then set the entrypoint to the path - i.e.

```json
"configSource": {
  "uri": "git+https://github.com/tektoncd/catalog",
  "digest": {
    "sha1": "15bf79ea9c89fffbf5dd02c6b5b686b291bfcbd2"
  },
  "entryPoint": "<kind>/<name>/<version>/<name>.yaml"
},
```

For a generic file based ref (e.g. if it was backed by a volume) you could just return the checksum of the file itself.

The important thing is a third-party (like chains) should be able to go back and verify the artifact integrity.

re Catalog UIs - in either case, both UIs will still be backed by the same catalog backend, right? So it shouldn't matter where this is being served from.

Reply:
> re Catalog UIs - in either case, both UIs will still be backed by the same catalog backend, right? So it shouldn't matter where this is being served from.

It actually matters: The Hub resolver will call Hub APIs to fetch the resource, not directly fetches from the catalog repo (like git resolver). For example, today in Hub resolver, it calls Tekton Hub API```https://api.hub.tekton.dev/v1/resource/tekton/pipeline/buildah/0.1/yaml```.  If the ```source``` is set to "Artifact Hub", another Artifact Hub API will be called.

So in order to have Digest returned by both Tekton Hub and Artifact Hub, we need to add such support in both Tekton Hub API (which will be deprecated) and Artifact Hub API

Reply:
For the Tekton Hub API, you could always add the digest in the resolver if you don't want to modify the API.

For ArtifactHub API, it looks like it already returns the sha256 sum. Separately, not sure how involved you are with the upstream ArtifactHub API, but it would be nice if we could upstream source provenance / attestation information in the API response. 👀 Happy to chat more off thread if you're interested!

Reply:
Thanks for sharing all thoughts here. Please include me in the off thread chat/meet as well :D haha love to learn more about this as well. 
Thank you ~~

Reply:
> 



> For the Tekton Hub API, you could always add the digest in the resolver if you don't want to modify the API.
> 
> For ArtifactHub API, it looks like it already returns the sha256 sum in the API response, so we should be good to go there.

> 
> Separately, not sure how involved you are with the upstream ArtifactHub API, but it would be nice if we could upstream source provenance / attestation information in the API response. 👀 Happy to chat more off thread if you're interested!

For Artifact Hub API: sure, if this digest works for this requirement then it is perfectly naturally supported :) What I thought was you need the commit sha

For the Tekton Hub API: if the checksum of the file is good then we don't need to modify the API (again  I thought what you need was the commit sha)

And yes, we are in the early stage of using Artifact Hub, we can definitely discuss more about source provenance / attestation :)

Reply:
Commit SHA would be nice, but not a hard requirement. :)

Would great to add more source provenance information into the ArifactHub API response long term though!